packages/contracts-bedrock/src/L2/L2ToL2CrossDomainMessenger.sol

```diff
@@ -99,19 +99,30 @@ contract L2ToL2CrossDomainMessenger is IL2ToL2CrossDomainMessenger, ISemver, Tra
     /// @param _destination Chain ID of the destination chain.
     /// @param _target      Target contract or wallet address.
     /// @param _message     Message payload to call target with.
-    function sendMessage(uint256 _destination, address _target, bytes calldata _message) external {
+    /// @return _msgHash The hash of the message being sent, which can be used for tracking whether
+    ///                  the message has successfully been relayed.
+    function sendMessage(
+        uint256 _destination,
+        address _target,
+        bytes calldata _message
+    )
+        external
+        returns (bytes32 _msgHash)
+    {
         if (_destination == block.chainid) revert MessageDestinationSameChain();
         if (_target == Predeploys.CROSS_L2_INBOX) revert MessageTargetCrossL2Inbox();
         if (_target == Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER) revert MessageTargetL2ToL2CrossDomainMessenger();
 
+        (uint256 _source, uint256 _messageNonce, address _sender) = (block.chainid, messageNonce(), msg.sender);
```

Comment:
I don't love the style here. The prefix of `_` means that it is the argument to a function. No variables should have that unless it's an argument to a function. It is sometimes used as the prefix for an `internal` function, but that is not consistently used. I see that it's broken out into its own line to cache env context onto the cache. I think that is fine but the variable names need to remove the prefix underscore